src/diff/component.js

```diff
@@ -45,12 +45,16 @@ export function renderFunctionComponent(
 	c.context = componentContext;
 	internal.props = c.props = newProps;
 
-	if ((tmp = options._render)) tmp(internal);
-
-	internal.flags &= ~DIRTY_BIT;
 	c._internal = internal;
-
-	tmp = type.call(c, c.props, c.context);
+	// eslint-disable-next-line
+	while (1) {
```

Comment:
Could L54 be merged into this loop condition?


```suggestion
	while (internal.flags & DIRTY_BIT) {
```

The dirty bit will be set on the first iteration, then the first line will unset it, and the loop will exit unless state is updated.

Reply:
Well components subject from trickle down updates won't be dirty as their state isn't set, they are just children from a state-updated parent which means they would never enter this clause

Reply:
This also needs a count check (react uses 25) to break an infinite loop.